### PR TITLE
[Mistweaver] Dancing Mist update

### PR DESCRIPTION
**Pull request by vohrr**
### Description
Updated the hot attributor and hot tracker to more accurately track dancing mists by applying the the properties of the hot it was copied from 
Updated dancing mist to show a breakdown of procs by trigger

### Screenshot(s):
image

### CODE REVIEW

File: src/analysis/retail/monk/mistweaver/normalizers/CastLinkNormalizer.ts [hunk 1]
```diff
@@ -36,7 +37,7 @@ export const SOOM_GOM = 'SoomGOM';
 export const VIVIFY = 'Vivify';
 
 const RAPID_DIFFUSION_BUFFER_MS = 300;
-const DANCING_MIST_BUFFER_MS = 120;
+const DANCING_MIST_BUFFER_MS = 140;
```
Comment (trevorm4) [inline src/analysis/retail/monk/mistweaver/normalizers/CastLinkNormalizer.ts:40]: Might need to be increased as discussed offline but I'll leave that up to you

File: src/analysis/retail/monk/mistweaver/modules/core/HotTrackerMW.tsx [hunk 2]
```diff
@@ -38,6 +38,24 @@ class HotTrackerMW extends HotTracker {
...
+    return hot.attributions.some(function (attr) {
+      return attr.name === ATTRIBUTION_STRINGS.DANCING_MIST_SOURCES.DM_SOURCE_HC;
+    });
+  }
+
+  dancingMist_MistsOfLife(hot: Tracker): boolean {
+    return hot.attributions.some(function (attr) {
+      return attr.name === ATTRIBUTION_STRINGS.DANCING_MIST_SOURCES.DM_SOURCE_MOL;
+    });
+  }
```
Comment (trevorm4) [inline src/analysis/retail/monk/mistweaver/modules/core/HotTrackerMW.tsx]: Not a huge fan of these function names, mixing styles is a bit messy imo

**Review by trevorm4 — APPROVED** [review]

File: src/analysis/retail/monk/mistweaver/normalizers/CastLinkNormalizer.ts (hunk 1 above)
Comment (vohrr) [reply src/analysis/retail/monk/mistweaver/normalizers/CastLinkNormalizer.ts:40]: I will do some more testing and make another PR if 250 looks better

File: src/analysis/retail/monk/mistweaver/modules/core/HotTrackerMW.tsx (hunk 2 above)
Comment (vohrr) [reply src/analysis/retail/monk/mistweaver/modules/core/HotTrackerMW.tsx]: renamed

**Pull request merged**